### ISSUE TITLE
Cannot get package list from jenkinsci/workflow-demo container

### ISSUE THREAD

**Issue by johnmccabe** [puppetlabs/lumogon#20]
The following error when attempting to query the installed packages on the `jenkinsci/workflow-demo` container.
```
"dpkg": {
    "$schema": "http://puppet.com/lumogon/capability/label/draft-01/schema#1",
    "harvestid": "de22dc9b-c179-4c64-97bb-4af702ccaeb0",
    "payload": {
        "error": "Bytes returned from stream [4088] does not match the length specified in the header [4096]"
    },
    "title": "Dpkg Capability",
    "type": "dockerapi"
},
```
Exec-ing into the container and running the `dpkg-query` is successful.

**Comment by johnmccabe** [comment]
This has been confirmed and reproduced - thx @ccaum / @sbeaulie for reporting

**Comment by johnmccabe** [comment]
This is a bug in `dockeradapter.FilterDockerStream()`, it doesn't handle the chunked reads correctly, either needs to use ReadAll or handle partial Reads.

**Comment by johnmccabe** [comment]
@sbeaulie if you get a chance, you should be able to `docker pull puppet/lumogon` to get the latest client with the fix above

**Comment by sbeaulie** [comment]
It works! 

![image](https://user-images.githubusercontent.com/16609499/26876301-761d5074-4b7d-11e7-8459-478a0842b761.png)

**Comment by johnmccabe** [comment]
This was addressed in #27

**Comment by jfryman** [comment]
🤘
